state/accountsDBApi.go

```diff
@@ -212,11 +212,19 @@ func (accountsDB *accountsDBApi) Close() error {
 
 // GetAccountWithBlockInfo returns the account and the associated block info
 func (accountsDB *accountsDBApi) GetAccountWithBlockInfo(address []byte) (vmcommon.AccountHandler, common.BlockInfo, error) {
-	blockInfo, err := accountsDB.recreateTrieIfNecessary()
+	_, err := accountsDB.recreateTrieIfNecessary()
 	if err != nil {
 		return nil, nil, err
 	}
 
+	// We hold the read mutex over both <getting the current block info> AND <getting the account>.
+	// -> desired side-effect: any concurrent "recreateTrieIfNecessary()" waits until the mutex is released.
+	// -> under normal circumstances (node already synchronized), performance of GET account should not be impacted.
+	accountsDB.mutBlockInfoOfRecreatedTrie.RLock()
```

Comment:
would it be worth it so that we have a `recreateTrieIfNecessaryUnprotected` and apply the mutex over the entire action?

Reply:
Holding a write mutex spanning the whole function would prevent concurrent reads from the same recreated trie. Holding a read mutex will not be sufficient, normally. A mix of the two would be possible (not trivial) - I wanted to keep changes minimal in this PR. Is it all right to postpone a refactoring to another PR, perhaps?